### PR TITLE
Add extensive note about "the indexing warning" in README

### PR DESCRIPTION
**Pull request by hemidactylus**
This adds a section about the reason, the origins and the suggested mitigations of the "catch-all indexing" warning that may arise when creating e.g. a vector store with a pre-existing collection created for example in the UI.

We periodically see questions on this very issue, so documenting it here would be a first step. My plan if this verbiage is accepted is to essentially get the same paragraph in LC's docs for greater visibility.

### CODE REVIEW

**Comment by hemidactylus** [comment]
For easier check, this is the link to the rendered paragraph: https://github.com/langchain-ai/langchain-datastax/blob/SL-indexing-note-on-readme/libs/astradb/README.md#warnings-about-indexing

File: libs/astradb/README.md [hunk 1]
```diff
@@ -100,3 +100,27 @@ store = AstraDBByteStore(
     token="AstraCS:...",
 )
 ```
+
+## Warnings about indexing
```
Comment (cbornet) [inline libs/astradb/README.md:104]: We could link to this section in the error returned ?

File: libs/astradb/README.md [hunk 2]
```diff
@@ -100,3 +100,27 @@ store = AstraDBByteStore(
...
+Keep in mind that this is a warning and your application will continue running just fine, as long as you don't store very long texts.
+Should you need to add to a vector store, for example, a `Document` whose `page_content` exceeds ~8K in length, you will receive an indexing error from the database.
+
+### Remediation
+
+You have several options:
+
+- you can ignore the warning because you know your application will never need to store very long textual contents;
+- you can ignore the warning and explicitly instruct the plugin _not to_ create the collection, assuming it exists already (which suppresses the warning): `store = AstraDBVectorStore(..., setup_mode=langchain_astradb.utils.astradb.SetupMode.OFF)`. In this case the collection will be used as-is, no (indexing) questions asked;
+- if you can afford re-populating the collection anew, you can drop it and re-run the LangChain application: the collection will be created with the optimized indexing settings. **This is the recommended option, when possible**.
```
Comment (cbornet) [inline libs/astradb/README.md:126]: "re-populating the collection anew": remove `re-` or `anew`

File: libs/astradb/README.md [hunk 3]
```diff
@@ -100,3 +100,27 @@ store = AstraDBByteStore(
...
+
+The reason for the warning is that the requested collection already exists on the database, and it is configured to [index all of its fields for search](https://docs.datastax.com/en/astra-db-serverless/api-reference/collections.html#the-indexing-option), possibly implicitly, by default. When the LangChain object tries to create it, it attempts to enforce, instead, an indexing policy tailored to the prospected usage. For example, the LangChain vector store will index the metadata but leave the textual content out: this is both to enable storing very long texts and to avoid indexing fields that will never be used in filtering a search (indexing those would also have a slight performance cost for writes).
+
+Typically there are two reasons why you may encounter the warning:
+
+1. you have created a collection by other means than letting the `AstraDBVectorStore` do it for you: for example, through the Astra UI, or using AstraPy's `create_collection` method of class `Database` directly;
```
Comment (cbornet) [inline libs/astradb/README.md:114]:
```suggestion
1. you have created a collection by other means than letting the `AstraDBVectorStore` do it for you: for example, through the Astra UI, or using AstraPy's `create_collection` method of class `Database` directly.
```

File: libs/astradb/README.md (hunk 1 above)
Comment (hemidactylus) [reply libs/astradb/README.md:104]: Awesome idea, done!

**Review by cbornet — APPROVED** [review]

**Pull request merged**